app/models/user.rb

```diff
@@ -993,7 +997,7 @@ def editable_fields_of_user(user)
     if user.wca_id.blank? && organizer_for?(user)
       fields << :name
     end
-    if admin? || any_kind_of_delegate? || results_team? || communication_team?
+    if can_edit_any_user?
```

Comment:
Can you please use this method in line 991 as well? Something like `user == self || can_edit_any_user?`